### DISCUSSION TITLE
What can one tell from error message like this?

### DISCUSSION THREAD

**Discussion by hput** [psy0rz/zfs_autobackup#143] [Q&A]
#### Synchronising
  [Target] p0/rep/2x/rpool@localsnpr-20220524191926: receiving full
! [Target] STDERR > internal error: Invalid argument
! [Target] STDERR > /bin/sh: 2031: Abort(coredump)
! [Target] Command "zfs recv -u -o canmount=noauto -v -s p0/rep/2x/rpool" returned exit code -6 (valid codes: [0])
! [Source] rpool: FAILED: Last command returned error
  [Target] p0/rep/2x/rpool/ROOT@localsnpr-20220524191926: receiving full
  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy@localsnpr-20220524191926: receiving full
  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy/spool@localsnpr-20220524191926: receiving full

I guess the error is in the rpool part of the rpool file systems?  I'm thinking the bit "bin/sh: 2031" means something at line 2031 of some bin/sh script... was involved.  But line 2031 of what?

**Comment by psy0rz** [comment]
its not a linenumber. the zfs recv command is actually crashing, which can have various reasons.

hopefully someone can assist. (its not an error generated by zfs-autobackup itself)

**Reply by hput** [reply]
Hitting something that looks similar again: In the output of another replication pull 
Anyway to tell what the "invalid argument" is?
And at the end it says dataset failed and quits  
Does it mean just the last one mentioned?   p0/rep/2x/rpool/USERDATA/root_449iia
I would really like to get a clean backup .... any one have an idea?
This command from backup server started the pull:
      zfs-autobackup -v --no-holds --clear-mountpoint --ssh-source 2x r2x.oi p0/rep/2x

[...]
  #### Synchronising
  [Target] p0/rep/2x/rpool@r2x.oi-20220525233021: receiving incremental
! [Target] STDERR > internal error: Invalid argument
! [Target] STDERR > /bin/sh: 1114: Abort(coredump)
! [Target] Command "zfs recv -u -o canmount=noauto -v -s p0/rep/2x/rpool" returned exit code -6 (valid codes: [0])
! [Source] rpool: FAILED: Last command returned error
  [Target] p0/rep/2x/rpool/ROOT@r2x.oi-20220525233021: receiving incremental
  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy@r2x.oi-20220525233021: receiving incremental

[...]

  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy/var/log@r2x.oi-20220525233021: receiving incremental
  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy/var/mail@r2x.oi-20220525233021: receiving incremental
  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy/var/snap@r2x.oi-20220525233021: receiving incremental
  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy/var/spool@r2x.oi-20220525233021: receiving incremental
  [Target] p0/rep/2x/rpool/ROOT/ubuntu_h55lfy/var/www@r2x.oi-20220525233021: receiving incremental
  [Target] p0/rep/2x/rpool/USERDATA@r2x.oi-20220525233021: receiving incremental
  [Target] p0/rep/2x/rpool/USERDATA/root_449iia@r2x.oi-20220525233021: receiving incremental
! 1 dataset(s) failed!

**Reply by hput** [reply]
A puzzling factor in this error is that output ends with:
   ! 1 dataset(s) failed!

Surely this must mean the one in the error which apparently is the base element rpool:

  ! [Target] Command "zfs recv -u -o canmount=noauto -v -s p0/rep   /2x/rpool" returned exit code -6 (valid codes: [0])
  ! [Source] rpool: FAILED: Last command returned error

But all the file systems in the rpool hierarchy are live and well on backup server... near as I can see

A fact I probably should have been including from the start is that the backup server is an openindiana host with solaris like OS.
Is it possible that openindian zfs doesn't play well ubuntu linux zfs (the --ssh-source)?

But if so, it seems there would be more errors.
